test/unit-test/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+# Include filepaths for source and include.
+include( ${MODULE_ROOT_DIR}/mqttFilePaths.cmake )
+
+# ====================  Define your project name (edit) ========================
+set(project_name "mqtt_agent")
+
+# =====================  Create your mock here  (edit)  ========================
+
+# list the files to mock here
+list(APPEND mock_list
+            "${MODULE_ROOT_DIR}/source/dependency/coreMQTT/source/include/core_mqtt.h"
+        )
+# list the directories your mocks need
+list(APPEND mock_include_list
+            .
+            ${CMAKE_CURRENT_LIST_DIR}/logging
+            ${MQTT_AGENT_INCLUDE_PUBLIC_DIRS}
```

Comment:
This doesn't seem like it has the correct includes because `core_mqtt.h` has a dependency on `core_mqtt_serializer.h` and `transport_interface.h`, which aren't part of these directories.